eZ/Publish/API/Repository/Tests/BaseURLServiceTest.php

```diff
@@ -45,11 +50,121 @@ protected function assertSearchResultItems(SearchResult $searchResult, array $ex
             $this->assertContains($contentInfo->id, $expectedContentInfoIds);
         }
     }
+
+    protected function createContentWithLink(string $name, string $url, string $languageCode = 'eng-GB', int $parentLocationId = 2): Content
+    {
+        $repository = $this->getRepository(false);
+        $contentService = $repository->getContentService();
+        $contentTypeService = $repository->getContentTypeService();
+        $locationService = $repository->getLocationService();
+
+        try {
+            $contentType = $contentTypeService->loadContentTypeByIdentifier(self::URL_CONTENT_TYPE_IDENTIFIER);
+        } catch (NotFoundException $e) {
+            $contentType = $this->createContentTypeWithUrl();
+        }
+
+        $struct = $contentService->newContentCreateStruct($contentType, $languageCode);
+        $struct->setField('name', $name, $languageCode);
+        $struct->setField('url', $url, $languageCode);
+
+        $contentDraft = $contentService->createContent(
+            $struct,
+            [$locationService->newLocationCreateStruct($parentLocationId)]
+        );
+
+        return $contentService->publishVersion($contentDraft->versionInfo);
+    }
+
+    private function createContentTypeWithUrl(): ContentType
+    {
+        $repository = $this->getRepository();
+
+        $contentTypeService = $repository->getContentTypeService();
+
+        $typeCreate = $contentTypeService->newContentTypeCreateStruct(self::URL_CONTENT_TYPE_IDENTIFIER);
+        $typeCreate->mainLanguageCode = 'eng-GB';
+        $typeCreate->urlAliasSchema = 'url|scheme';
+        $typeCreate->nameSchema = 'name|scheme';
+        $typeCreate->names = [
+            'eng-GB' => 'URL: ' . self::URL_CONTENT_TYPE_IDENTIFIER,
+        ];
+        $typeCreate->descriptions = [
+            'eng-GB' => '',
+        ];
+        $typeCreate->creatorId = $this->generateId('user', $repository->getPermissionResolver()->getCurrentUserReference()->getUserId());
+        $typeCreate->creationDate = $this->createDateTime();
+
+        $nameFieldCreate = $contentTypeService->newFieldDefinitionCreateStruct('name', 'ezstring');
+        $nameFieldCreate->names = [
+            'eng-GB' => 'Name',
+        ];
+        $nameFieldCreate->descriptions = [
+            'eng-GB' => '',
+        ];
+        $nameFieldCreate->fieldGroup = 'default';
+        $nameFieldCreate->position = 1;
+        $nameFieldCreate->isTranslatable = false;
+        $nameFieldCreate->isRequired = true;
+        $nameFieldCreate->isInfoCollector = false;
+        $nameFieldCreate->validatorConfiguration = [
+            'StringLengthValidator' => [
+                'minStringLength' => 0,
+                'maxStringLength' => 0,
+            ],
+        ];
+        $nameFieldCreate->fieldSettings = [];
+        $nameFieldCreate->isSearchable = true;
+        $nameFieldCreate->defaultValue = '';
+
+        $typeCreate->addFieldDefinition($nameFieldCreate);
```

Comment:
Fields creation could be extracted to private methods to make this method more clear